docs/reference/indices/put-index-template.asciidoc

```diff
@@ -0,0 +1,335 @@
+
+[source,console]
+--------------------------------------------------
+PUT /_index_template/template_1
+{
+  "index_patterns" : ["t*"],
+  "priority" : 0,
+  "template": {
+    "settings" : {
+      "number_of_shards" : 1,
+      "number_of_replicas": 0
+    },
+    "mappings" : {
+      "_source" : { "enabled" : false }
+    }
+  }
+}
+
+PUT /_index_template/template_2
+{
+  "index_patterns" : ["te*"],
+  "priority" : 1,
+  "template": {
+    "settings" : {
+      "number_of_shards" : 2
+    },
+    "mappings" : {
+      "_source" : { "enabled" : true }
+    }
+  }
+}
+--------------------------------------------------
+
+For indices that start with `te*`, `_source` will enabled, and the index will have two primary
+shards and one replica, because only `template_2` will be applied.
+
+NOTE: Multiple templates with overlapping index patterns at the same priority are not allowed, and
+an error will be thrown when attempting to create a template matching an existing index template at
+identical priorities.
+
+
+[[versioning-templates]]
+===== Template versioning
+
+You can use the `version` parameter to add an optional version number to an index template. External
+systems can use these version numbers to simplify template management.
+
+The `version` parameter is completely optional and not automatically generated by {es}.
+
+To unset a `version`, replace the template without specifying one.
+
+[source,console]
+--------------------------------------------------
+PUT /_index_template/template_1
+{
+  "index_patterns" : ["foo", "bar"],
+  "priority" : 0,
+  "template": {
+    "settings" : {
+        "number_of_shards" : 1
+    }
+  },
+  "version": 123
+}
+--------------------------------------------------
+
+To check the `version`, you can use the <<indices-get-template, get index template>> API.
+
+[[template-metadata]]
+===== Template metadata
+
+You can use the `_meta` parameter to add optional metadata to an index template. This is a
+user-defined map that can contain any data. This data will be stored in the cluster state however,
+so keeping it short is preferrable.
+
+The `_meta` parameter is completely optional and not automatically generated by {es}.
+
+To unset `_meta`, replace the template without specifying one.
+
+[source,console]
+--------------------------------------------------
+PUT /_index_template/template_1
+{
+  "index_patterns": ["foo", "bar"],
+  "template": {
+    "settings" : {
+        "number_of_shards" : 3
+    }
+  },
+  "_meta": {
+    "description": "set number of shards to three",
+    "serialization": {
+      "class": "MyIndexTemplate",
+      "id": 17
+    }
+  }
+}
+--------------------------------------------------
+
+To check the `_meta`, you can use the <<indices-get-template, get index template>> API.
+
+[[data-stream-definition]]
+===== Data stream definition
+
+If an index template should create data streams, the template must include an
+empty `data_stream` object. See <<create-a-data-stream-template>>.
```

Comment:
It may be helpful to mention that these index templates cannot be used for regular indices.